Front-End/challenge2/script.js

```diff
@@ -5,3 +5,104 @@ TO-DO:
 - The calculator should be completely functional
 
 */
+
+"use strict"
+
+const wrapper = document.querySelector('.wrapper')
+
+let numberStr = ''
+let viewString = ''
+let operations = ['+', '-', 'X', '/']
+let stack = []
+
+const updateValue = (value) => {
+    document.getElementById('display').textContent = value;
+}
+
+const evaluateStack = (arrStack) => {
+
+    //Convertimos a Notacion Posfija
+    if (arrStack.length == 0)
+        return 0;
+
+    let auxStack = []
+    let lastOp = '$';
+    for (let i = 0; i < arrStack.length; i++) {
+        if (operations.includes(arrStack[i])) {
+            lastOp = arrStack[i];
+        } else {
+            auxStack.push(arrStack[i])
+            if (lastOp != '$') {
+                auxStack.push(lastOp)
+            }
+        }
+    }
+    console.log(auxStack)
+    
+
+
+    let mainStack = []
+    let last;
+    mainStack.push(parseFloat(auxStack[0]));
+
+    for (let i = 1; i < auxStack.length; i++) {
+        let current = auxStack[i]
+        if (!operations.includes(current)) {
+            mainStack.push(current)
+        } else {
+            switch (current) {
+                case '+':
+                    last = mainStack.pop()
```

Comment:
it seems we are repeating the same code in all the case statements, in that case it is better to abstract this logic to another function that you can reuse

Reply:
maybe another option is to evaluate something like `['+', '-', '/'].includes(current)` and if this is true do your operation, that way you avoid duplicating code and having a case for each operator